common/src/main/java/com/revenuecat/purchases/common/networking/ETagManager.kt

```diff
@@ -2,47 +2,66 @@ package com.revenuecat.purchases.common.networking
 
 import android.content.Context
 import android.content.SharedPreferences
+import androidx.annotation.VisibleForTesting
 import com.revenuecat.purchases.VerificationResult
+import com.revenuecat.purchases.common.DateProvider
+import com.revenuecat.purchases.common.DefaultDateProvider
 import com.revenuecat.purchases.common.LogIntent
 import com.revenuecat.purchases.common.log
 import com.revenuecat.purchases.strings.NetworkStrings
 import org.json.JSONObject
 import java.util.Date
 
-private const val SERIALIZATION_NAME_ETAG = "eTag"
-private const val SERIALIZATION_NAME_HTTPRESULT = "httpResult"
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+data class ETagData(
+    val eTag: String,
+    val lastRefreshTime: Date?
+)
 
+@VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
 data class HTTPResultWithETag(
-    val eTag: String,
+    val eTagData: ETagData,
     val httpResult: HTTPResult
 ) {
     fun serialize(): String {
         return JSONObject().apply {
-            put(SERIALIZATION_NAME_ETAG, eTag)
+            put(SERIALIZATION_NAME_ETAG, eTagData.eTag)
+            eTagData.lastRefreshTime?.let { put(SERIALIZATION_NAME_LAST_REFRESH_TIME, it.time) }
             put(SERIALIZATION_NAME_HTTPRESULT, httpResult.serialize())
         }.toString()
     }
 
     companion object {
+        private const val SERIALIZATION_NAME_ETAG = "eTag"
+        private const val SERIALIZATION_NAME_LAST_REFRESH_TIME = "lastRefreshTime"
+        private const val SERIALIZATION_NAME_HTTPRESULT = "httpResult"
+
         fun deserialize(serialized: String): HTTPResultWithETag {
             val jsonObject = JSONObject(serialized)
             val eTag = jsonObject.getString(SERIALIZATION_NAME_ETAG)
+            val lastRefreshTime = jsonObject.optLong(SERIALIZATION_NAME_LAST_REFRESH_TIME, -1L)
+                .takeIf { it != -1L }
+                ?.let { Date(it) }
             val serializedHTTPResult = jsonObject.getString(SERIALIZATION_NAME_HTTPRESULT)
-            return HTTPResultWithETag(eTag, HTTPResult.deserialize(serializedHTTPResult))
+            return HTTPResultWithETag(ETagData(eTag, lastRefreshTime), HTTPResult.deserialize(serializedHTTPResult))
         }
     }
 }
 
 class ETagManager(
-    private val prefs: SharedPreferences
+    private val prefs: SharedPreferences,
+    private val dateProvider: DateProvider = DefaultDateProvider()
 ) {
 
-    internal fun getETagHeader(
+    internal fun getETagHeaders(
         path: String,
         refreshETag: Boolean = false
-    ): Map<String, String> {
-        val eTagHeader = HTTPRequest.ETAG_HEADER_NAME to if (refreshETag) "" else getETag(path)
-        return mapOf(eTagHeader)
+    ): Map<String, String?> {
+        val eTagData = if (refreshETag) null else getETagData(path)
+        return mapOf(
+            HTTPRequest.ETAG_HEADER_NAME to eTagData?.eTag.orEmpty(),
+            HTTPRequest.ETAG_LAST_REFRESH_NAME to eTagData?.lastRefreshTime?.time?.toString(),
```

Comment:
If `lastRefreshTime` is null, this will send a null value. Do we want that? Or is it better to not send the header at all

Reply:
In the caller, we filter headers with null values so we wouldn't send this header if it's null. I thought about sending an empty value like we do for the ETAG but I do think it's preferred to not send the header at all.